File: src/types/disease.ts
// Matches the backend DiseasePredictionResponse schema
export interface DiseaseResult {
  crop: string;
  disease_detected: string;
  confidence: number;
  severity: string;
  treatment: string[];
  status: string;
  image_source: string;
  recommended_action: string;
}

export interface DiseaseHistoryItem {
  id: number;
  crop: string;
  disease_detected: string;
  confidence: number;
  severity: string;
  image_source: string;
  image_url?: string | null;
  created_at: string;
}

File: backend/app/models/disease.py
from sqlalchemy import Column
from sqlalchemy import Integer
from sqlalchemy import Float
from sqlalchemy import String
from sqlalchemy import DateTime

from datetime import datetime

from app.database.database import Base


class DiseaseDetection(Base):
    __tablename__ = "disease_detections"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    crop = Column(String)

    disease_detected = Column(String)

    confidence = Column(Float)

    severity = Column(String)

    treatment = Column(String)

    image_source = Column(String)

    image_filename = Column(String, nullable=True)

    created_at = Column(
        DateTime,
        default=datetime.utcnow
    )